pregen.js

```diff
@@ -91,3 +96,76 @@ export function preparePythonEnv(filePath, options) {
     }
   }
 }
+
+/**
+ * Method to check and prepare the environment for node
+ *
+ * @param {String} filePath Path
+ * @param {Object} options CLI Options
+ */
+export function prepareNodeEnv(filePath, options) {
+  // if (hasAnyProjectType("node", options, false)) {
+  //   if (arch() !== "x64") {
+  //     console.log(
+  //       `INFO: Many pypi packages have poor support for ${arch()} architecture.\nRun the cdxgen container image with --platform=linux/amd64 for best experience.`,
+  //     );
+  //   }
+  //   if (platform() === "win32") {
+  //     console.log(
+  //       "Install the appropriate compilers and build tools on Windows by following this documentation - https://wiki.python.org/moin/WindowsCompilers",
+  //     );
+  //   }
+  // }
+
+  for (const pt of options.projectType) {
+    if (pt.startsWith("node") && !process.env.NODE_INSTALL_ARGS) {
+      if (!isNvmAvailable()) {
+        console.log(
+          "Install nvm by following the instructions at https://github.com/nvm-sh/nvm",
+        );
+        return;
+      }
+      const nodeVersion = pt.replace(/\D/g, "");
+      installNvmTool(nodeVersion);
+      doNpmInstall(filePath, nodeVersion);
+    }
+  }
+}
+
+/**
+ * This method installs and create package-lock.json
+ *
+ * @param {String} filePath Path
+ * @param {String} nodeVersion number
+ */
+export function doNpmInstall(filePath, nodeVersion) {
+  // we do not install if INSTALL_ARGS set false
+  if (process.env.NODE_INSTALL_ARGS === false) {
+    return;
+  }
+  const currentDir = process.cwd();
```

Comment:
Pass cwd to spawnSync instead of changing the entire process. Since node is a single process event threaded, it will affect other parts of the code.